src/state/charts/hooks.js

```diff
@@ -0,0 +1,118 @@
+};
+
+export const usePowerBIInfo = () => {
+  return useSelector((state) => state.charts?.powerbi);
+};
+
+export const usePowerBIReducerStatus = () => {
+  return useSelector((state) => state.charts?.powerbi?.status);
+};
+
+export const usePowerBIReportsConfig = () => {
+  return useSelector((state) => state.charts?.powerbi?.data?.reportsConfig);
+};
+
+export const useDashboardsViewReportsConfig = () => {
+  return useSelector((state) => state.charts?.powerbi?.data?.reportsConfig?.dashboardsView) ?? [];
+};
+
+export const useScenarioViewReportsConfig = () => {
+  return useSelector((state) => state.charts?.powerbi?.data?.reportsConfig?.scenarioView) ?? [];
+};
+
+export const useCurrentScenarioReportConfig = () => {
+  const currentScenarioData = useCurrentSimulationRunnerData();
+  const scenarioViewReportsConfig = useScenarioViewReportsConfig();
+  return useMemo(
+    () => PowerBIUtils.getScenarioViewReportConfig(scenarioViewReportsConfig, currentScenarioData?.runTemplateId),
+    [scenarioViewReportsConfig, currentScenarioData?.runTemplateId]
+  );
+};
+
+export const usePowerBIReports = () => {
+  return useSelector((state) => state.charts?.powerbi?.data?.reports);
+};
+
+export const useSetPowerBIReportsConfig = (reportsConfig) => {
+  const dispatch = useDispatch();
+  return useCallback((scenario) => dispatch(setPowerBIReportConfig(scenario)), [dispatch]);
+};
+
+export const useSupersetInfo = () => {
+  return useSelector((state) => state.charts?.superset);
+};
+
+export const useSupersetReducerStatus = () => {
+  return useSelector((state) => state.charts?.superset?.status);
+};
+
+export const useSupersetGuestToken = () => {
+  return useSelector((state) => state.charts?.superset?.data?.guestToken);
+};
+
+export const useSupersetDashboards = () => {
+  return useSelector((state) => state.charts?.superset?.data?.dashboards);
+};
+
+export const useSupersetUrl = () => {
+  return useSelector((state) => state.charts?.superset?.data?.supersetUrl);
+};
+
+export const useCurrentSupersetDashboard = () => {
+  const currentScenarioData = useCurrentSimulationRunnerData();
+  const dashboards = useSupersetDashboards();
+
+  return useMemo(() => {
+    if (!dashboards?.scenarioView) {
+      return null;
+    }
+
+    const scenarioView = dashboards.scenarioView;
+    const runTemplateId = currentScenarioData?.runTemplateId;
+
+    let dashboardId = null;
+
+    if (Array.isArray(scenarioView)) {
+      const config = scenarioView.find((c) => c.runTemplateIds?.includes(runTemplateId));
+      dashboardId = config?.dashboardId;
+    } else if (typeof scenarioView === 'object') {
+      if (scenarioView.overrides?.[runTemplateId]) {
+        dashboardId = scenarioView.overrides[runTemplateId];
+      } else if (scenarioView.default) {
+        dashboardId = scenarioView.default;
+      }
+    }
+
+    if (!dashboardId) {
+      return null;
+    }
+
+    const fullDashboard = dashboards.dashboards?.find((d) => d.id === dashboardId);
```

Comment:
Same as above. We try to avoid simple letter as variables name. Maybe use `dashboard` instead of `d`